src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Windows.cs

```diff
@@ -36,6 +42,208 @@ internal IntPtr[] GetWaitedHandleArray(int requiredCapacity)
             return _waitedHandles.Items;
         }
 
+        private void PlatformSpecificInitializeExistingThread()
+        {
+            _osHandle = GetOSHandleForCurrentThread();
+            _priority = MapFromOSPriority((OSThreadPriority)Interop.mincore.GetThreadPriority(_osHandle));
+        }
+
+        private static SafeWaitHandle GetOSHandleForCurrentThread()
+        {
+            SafeWaitHandle threadHandle;
+            if (Interop.mincore.GetCurrentThreadHandle(out threadHandle))
+            {
+                return threadHandle;
+            }
+
+            // Throw an ApplicationException for compatibility with CoreCLR
+            var ex = new ApplicationException();
+            ex.SetErrorCode((int)Interop.mincore.GetLastError());
```

Comment:
mincore.GetLastError could have been overwritten by now. You should use Marshal.GetLastWin32Error